EIPS/eip-145.md

```diff
@@ -79,7 +79,282 @@ The newly introduced instructions have no effect on bytecode created in the past
+11. ```
+    PUSH 0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
+    PUSH 0x01
+    SHL
+    ---
+    0xfffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe
+    ```
+    
+
+### `SHR` (logical shift right)
+
+1. ```
+   PUSH 0x0000000000000000000000000000000000000000000000000000000000000001
+   PUSH 0x00
+   SHR
+   ---
+   0x0000000000000000000000000000000000000000000000000000000000000001
+   ```
+2. ```
+   PUSH 0x0000000000000000000000000000000000000000000000000000000000000001
+   PUSH 0x01
+   SHR
+   ---
+   0x0000000000000000000000000000000000000000000000000000000000000000
+   ```
+3. ```
+   PUSH 0x8000000000000000000000000000000000000000000000000000000000000000
+   PUSH 0x01
+   SHR
+   ---
+   0x4000000000000000000000000000000000000000000000000000000000000000
+   ```
+4. ```
+   PUSH 0x8000000000000000000000000000000000000000000000000000000000000000
+   PUSH 0xff
+   SHR
+   ---
+   0x0000000000000000000000000000000000000000000000000000000000000001
+   ```
+5. ```
+   PUSH 0x8000000000000000000000000000000000000000000000000000000000000000
+   PUSH 0x0100
+   SHR
+   ---
+   0x0000000000000000000000000000000000000000000000000000000000000000
+   ```
+6. ```
+   PUSH 0x8000000000000000000000000000000000000000000000000000000000000000
+   PUSH 0x0101
+   SHR
+   ---
+   0x0000000000000000000000000000000000000000000000000000000000000000
+   ```
+7. ```
+   PUSH 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
+   PUSH 0x00
+   SHR
+   ---
+   0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
+   ```
+8. ```
+   PUSH 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
+   PUSH 0x01
+   SHR
+   ---
+   0x7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
+   ```
+9. ```
+   PUSH 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
+   PUSH 0xff
+   SHR
+   ---
+   0x0000000000000000000000000000000000000000000000000000000000000001
+   ```
+10. ```
+    PUSH 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff
+    PUSH 0x0100
+    SHR
+    ---
+    0x0000000000000000000000000000000000000000000000000000000000000000
+    ```
+11. ```
+    PUSH 0x0000000000000000000000000000000000000000000000000000000000000000
+    PUSH 0x01
+    SHR
+    ---
+    0x0000000000000000000000000000000000000000000000000000000000000000
+    ```
+
+### `SAR` (arithmetic shift right)
+
+1. ```
+   PUSH 0x0000000000000000000000000000000000000000000000000000000000000001
+   PUSH 0x00
+   SAR
+   ---
+   0x0000000000000000000000000000000000000000000000000000000000000001
+   ```
+2. ```
+   PUSH 0x0000000000000000000000000000000000000000000000000000000000000001
+   PUSH 0x01
+   SAR
+   ---
+   0x0000000000000000000000000000000000000000000000000000000000000000
+   ```
+3. ```
+   PUSH 0x8000000000000000000000000000000000000000000000000000000000000000
+   PUSH 0x01
+   SAR
+   ---
+   0x12000000000000000000000000000000000000000000000000000000000000000
```

Comment:
I guess this shouldn't be like this.  The most significant bit turned from 1 to 0.

Reply:
oopsy. It should be 12 hex, i.e. 0xc